File: frontend/src/types/course.ts
export interface Course {
  id: number;
  title: string;
  description: string | null;
  youtube_url: string | null;
  thumbnail_url: string | null;
  total_duration: number | null;
  created_by: number | null;
  created_at: string | null;
  lessons?: Lesson[];
}

export interface Lesson {
  id: number;
  course_id: number;
  title: string;
  description: string | null;
  order_index: number;
  video_url: string | null;
  duration: number | null;
  summary: string | null;
  phase: string | null;
  created_at: string | null;
}

File: backend/app/models/course.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.models.base import Base


class Course(Base):
    __tablename__ = "courses"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    description = Column(Text)
    created_by = Column(Integer, ForeignKey("users.id"), index=True)
    youtube_url = Column(String)
    thumbnail_url = Column(String)
    total_duration = Column(Integer)  # seconds
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    creator = relationship("User", back_populates="courses")
    lessons = relationship(
        "Lesson",
        back_populates="course",
        order_by="Lesson.order_index",
        cascade="all, delete-orphan",
    )
